### DISCUSSION TITLE
[Feature Request] Force HDR on/off per client

### DISCUSSION THREAD

**Discussion by NobQuaffer** [ClassicOldSong/moonlight-android#23] [Ideas]
I stream to various devices, some support HDR and some don't. It would be nice to force HDR on/off on a per-client basis, e.g. HDR to my TV and phone, SDR to my G Cloud. I haven't had much luck getting it to automatically do so when using virtual display mode, it seems to always enable HDR regardless of the client.

My main PC display is HDR, so HDR is enabled in Windows 11, as is AutoHDR.

**Comment by ClassicOldSong** [comment]
Enable HDR on your client, and set "Follow Client's HDR config" in `Audio/Video` tab. This feature is [Apollo](https://github.com/ClassicOldSong/Apollo) only.

![image](https://github.com/user-attachments/assets/a3831527-44b8-48fa-a69f-30a8656048a4)

**Reply by NobQuaffer** [reply]
Sorry, only just seen your reply! I know I can tell it to follow the client's HDR config but that doesn't always seem to work. For example, I have an XBOX One Series S hooked up to an SDR TV and whenever I stream to it, it tries to run in HDR, which looks washed out and terrible on the SDR TV. 

Having the ability to force HDR/SDR for a specific client would be helpful for situations like this. It's nice that the software is generally smart enough to detect this but having the ability to override this ability couldn't hurt and I'm guessing would be easier than determining why it's attempting to use HDR for a client that is in SDR. My guess would be it's because the Xbox One Series S supports HDR, even though the TV it's attached to doesn't, but that's just a guess/theory.

**Reply by ClassicOldSong** [reply]
There're reports saying the Xbox client always request HDR regardless of the settings.

But you just need to disable the follow client HDR option and then disable it in Windows settings while streaming. It'll be remembered.

**Reply by NobQuaffer** [reply]
Thanks for getting back to me :) isn't that a universal setting? The Xbox is the only client I have that doesn't have a HDR display connected to it, all my other clients use HDR, as does my computer monitor. So, would that mean that I'd have to connect the stream for each of my HDR-compatible devices and manually set them to HDR or would HDR be assumed by default and thus I'd just need to disable it for the Xbox?

**Reply by ClassicOldSong** [reply]
If they're previously enabled then there's no need to enable them again, but it's just one simple settings that only need to be toggled once per client, I think that's fine.
